### PR TITLE
Fixes #445 - Ship Fenix x86_64 builds

### LINKED ISSUE #445: Ship Fenix x86_64 builds
We need to publish 64-bit builds to the Google Play Store before Google's August 1 deadline requiring 64-bit support (for apps that include native code).

https://android-developers.googleblog.com/2019/01/get-your-apps-ready-for-64-bit.html

It looks like Fenix is only shipping 32-bit ARMv7 and x86 builds:
https://github.com/mozilla-mobile/fenix/blob/19dd46863980808c0832cd8172a33d36b1ac29d0/.taskcluster.yml#L92-L93

GeckoView publishes [x86_64 builds in Maven](https://maven.mozilla.org/?prefix=maven2/org/mozilla/geckoview/geckoview-nightly-x86_64/), _but_ we haven't actually shipped x86_64 builds to Fennec Nightly users yet. I'm working with Releng and Relman to ship x86_64 Fennec Nightly by the end of this week.

Fenix will need to ship x86_64 builds by Google's August 1 deadline (or take down our 32-bit x86 builds), but I recommend that you wait for Fennec Nightly users to dogfood x86_64 for a couple weeks before shipping x86_64 Fenix builds. The beginning of Fenix M3 (March) would probably be a good time.

(For shipping ARM64 Fenix builds, see issue #444.)

### PR DESCRIPTION
**Pull request by st3fan**
This patch add support for building x86_64 builds. This is an experiment and I would like to get this working and tested on this PR branch first. This has low priority but ideally starts working before the end of July because of a Google Play Store deadline regarding 64-bit builds.

### CODE REVIEW

**Comment by boek** [comment]
Looks like the builds aren’t happy with it though

**Comment by st3fan** [comment]
@boek sorry i should have marked this as Do Not Merge or WIP - This has no priority, just something I'd like to get going.

**Comment by sarentz** [comment]
Askin for help from @mozilla-mobile/releng - TaskCluster tells me

> HTTP 424 Artifact "public/actions.json" not found at "/opt/fenix/actions.json"

And does not show any more details for these builds. Did I break things completely with my changes or is there something else going on?

**Comment by JohanLorenzo** [comment]
@st3fan hi! This Taskcluster error is some red herring[1]. In order to see the real error:

1. Copy the "Task Group ID" to the "Task ID" field and click on "Inspect"
2. Access the logs of the task.

In this case, it's a gradle error: `No such property: geckoview_beta_x86 for class: Gecko`

https://tools.taskcluster.net/groups/W7lE52rJSfaSQRSJNBeQvQ/tasks/W7lE52rJSfaSQRSJNBeQvQ/runs/0/logs/public%2Flogs%2Flive_backing.log#L392

[1] It shows up because the Taskcluster WebUI looks into this artifact to enable/disable commands under the "Actions" drop down. I've seen it for long, I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1554599 to track this.

**Comment by sarentz** [comment]
This now produces x86_64 APKs.

**Comment by pocmo** [comment]
> Without changes that still happens for x86_64. Or does x86_64 need special treatment?

They still need to be unique for every APK. And since the x86 AND x86_64 build can be installed on 64 bit x86 devices we would want to use a higher version code for the 64 bit build so that those devices get that build.

TL;DR: Adding a case that does `+3` for x86_64 should be enough here.

**Comment by sarentz** [comment]
I am just not having a lot of luck ...

```
> Task :architecture:compileDebugKotlin
Could not perform incremental compilation: Could not connect to Kotlin compile daemon
Could not connect to kotlin daemon. Using fallback strategy.

OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000b2150000, 82509824, 0) failed; error='Cannot allocate memory' (errno=12)

> Task :architecture:compileDebugKotlin FAILED
#
# There is insufficient memory for the Java Runtime Environment to continue.
# Native memory allocation (mmap) failed to map 82509824 bytes for committing reserved memory.
# An error report file with more information is saved as:
# /opt/fenix/hs_err_pid338.log

FAILURE: Build failed with an exception.
```

I'll run this again - but running out of memory is not something I've seen before.

File: app/build.gradle [hunk 1]
```diff
@@ -133,7 +139,9 @@ android.applicationVariants.all { variant ->
         // Our x86 builds need a higher version code to avoid installing ARM builds on an x86 device
         // with ARM compatibility mode.
 
-        if (variant.flavorName.contains("x86")) {
+        if (variant.flavorName.contains("x86_x86")) {
```
Comment (mitchhentges) [inline app/build.gradle]: `x86_64` here, I think

Comment (mitchhentges) [inline app/build.gradle]: `x86_64` here, I think

File: app/build.gradle [hunk 2]
```diff
@@ -133,7 +139,9 @@ android.applicationVariants.all { variant ->
...
+            versionCode = versionCode + 3
+        } else if (variant.flavorName.contains("x86")) {
```
Comment (mitchhentges) [inline app/build.gradle]: Hmm, this will match `x86_64`, too. :thinking: 
I guess the `if/else` saves us here, but it would be nice if we didn't depend on ordering

File: app/build.gradle (hunk 1 above)
Comment (sarentz) [reply app/build.gradle]: Duh :-(

File: app/build.gradle (hunk 2 above)
Comment (sarentz) [reply app/build.gradle]: It will never hit that line - this is pretty normal. I can do something way more complicated by parsing the flavorName but I don't know if that is worth it - this logic is pretty solid.

**Comment by sarentz** [comment]
@mitchhentges i see you editing in the `decision_task.py` too - do these changes still make sense? For example I see that I don't have to specify the architectures manually anymore - will it just discover by itself what flavors we have now?

**Comment by mitchhentges** [comment]
Looks great, yeah - it'll automatically detect all build types now for release

**Comment by sarentz** [comment]
@mitchhentges great - lets see if the last small changes result in a green build and then we can land it I hope.
